logic/src/commonTest/kotlin/com/wire/kalium/logic/data/call/CallRepositoryTest.kt

```diff
@@ -41,7 +45,175 @@ class CallRepositoryTest {
+
+        val calls = callRepository.getAllCalls()
+        assertEquals(1, calls.value.size)
+        assertEquals(calls.value[0].status, CallStatus.ESTABLISHED)
+    }
+
+    @Test
+    fun givenNoIncomingCallsInTheFlow_whenGetIncomingCallsIsCalled_thenReturnAnEmptyListInTheFlow() = runTest {
+        val calls = callRepository.getIncomingCalls()
+
+        assertEquals(0, calls.first().size)
+    }
+
+    @Test
+    fun givenSomeIncomingCallsInTheFlow_whenGetIncomingCallsIsCalled_thenReturnTheListOfIncomingCallsInTheFlow() = runTest {
+        callRepository.updateCallProfileFlow(
+            CallProfile(
+                mapOf(
+                    startedCall.conversationId.toString() to startedCall,
+                    answeredCall.conversationId.toString() to answeredCall,
+                    incomingCall.conversationId.toString() to incomingCall
+                )
+            )
+        )
+
+        val calls = callRepository.getIncomingCalls()
+
+        assertEquals(1, calls.first().size)
+        assertEquals(calls.first()[0], incomingCall)
+    }
+
+    @Test
+    fun givenNoOngoingCallsInTheFlow_whenGetOngoingCallIsCalled_thenReturnAnEmptyListInTheFlow() = runTest {
+        val calls = callRepository.getIncomingCalls()
+
+        assertEquals(0, calls.first().size)
+    }
+
+    @Test
+    fun givenSomeOngoingCallsInTheFlow_whenGetOngoingCallIsCalled_thenReturnTheListOfOngoingCallsInTheFlow() = runTest {
+        callRepository.updateCallProfileFlow(
+            CallProfile(
+                mapOf(
+                    startedCall.conversationId.toString() to startedCall,
+                    answeredCall.conversationId.toString() to answeredCall,
+                    incomingCall.conversationId.toString() to incomingCall,
+                    establishedCall.conversationId.toString() to establishedCall
+                )
+            )
+        )
+
+        val calls = callRepository.getOngoingCall()
+
+        assertEquals(2, calls.first().size)
+        assertEquals(calls.first()[0], answeredCall)
+        assertEquals(calls.first()[1], establishedCall)
+    }
+
     private companion object {
         const val CALL_CONFIG_API_RESPONSE = "{'call':'success','config':'dummy_config'}"
+        private const val randomConversationIdString = "random@domain"
+        private val conversationIdStartedCall = ConversationId("value1", "domain1")
+        private val conversationIdAnsweredCall = ConversationId("value2", "domain2")
+        private val conversationIdIncomingCall = ConversationId("value3", "domain3")
+        private val conversationIdEstablishedCall = ConversationId("value4", "domain4")
+
+        private val startedCall = Call(
+            conversationId = conversationIdStartedCall,
+            status = CallStatus.STARTED,
+            callerId = "caller-id",
+            participants = listOf(),
+            maxParticipants = 0
+        )
+
+        private val answeredCall = Call(
```

Comment:
A lot of duplicated lines in this file.

We could have a:

```kotlin

fun procideCall(id: ConversationId, status: CallStatus) = Call(
    id, 
    status, 
    callerId = "caller-id",
    participants = listOf(),
    maxParticipants = 0
)

```

And call this to create the different calls.